packages/container/lib/container.js

```diff
@@ -278,7 +281,7 @@ function isFactoryInstance(container, fullName, { instantiate, singleton }) {
 }
 
 function instantiateFactory(container, fullName, options) {
-  let factoryManager = EMBER_MODULE_UNIFICATION && options && options.source ? container.factoryFor(fullName, options) : container.factoryFor(fullName);
+  let factoryManager = EMBER_MODULE_UNIFICATION && options ? container.factoryFor(fullName, options) : container.factoryFor(fullName);
```

Comment:
I think we can just pass `container.factoryFor(fullName, options)` all the time.

Reply:
Turns out some tests related to local lookup fail if we remove this.